From d71d462cebdc0a4dd20bf0c79c7d3daf03d1ce26 Mon Sep 17 00:00:00 2001
From: Hendrik Borghorst <hendrikborghorst@gmail.com>
Date: Sat, 31 Jul 2021 08:17:28 +0000
Subject: [PATCH 1/3] ODROID_GO2: Enable Wifi GPIO toggle

Signed-off-by: Hendrik Borghorst <hendrikborghorst@gmail.com>
---
 arch/arm/mach-rockchip/board.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/board.c b/arch/arm/mach-rockchip/board.c
index 5304eb055c..cb276e32dc 100644
--- a/arch/arm/mach-rockchip/board.c
+++ b/arch/arm/mach-rockchip/board.c
@@ -22,6 +22,10 @@
 
 DECLARE_GLOBAL_DATA_PTR;
 
+#ifdef CONFIG_TARGET_ODROID_GO2
+#define WIFI_EN_GPIO   110 /* GPIO3_B6 */
+#endif
+
 __weak int rk_board_late_init(void)
 {
 	return 0;
@@ -30,13 +34,17 @@ __weak int rk_board_late_init(void)
 int board_late_init(void)
 {
 	setup_boot_mode();
-
 	return rk_board_late_init();
 }
 
 int board_init(void)
 {
 	int ret;
+#ifdef CONFIG_TARGET_ODROID_GO2
+       gpio_request(WIFI_EN_GPIO, "wifi_enable");
+       gpio_direction_output(WIFI_EN_GPIO, 1);
+       gpio_free(WIFI_EN_GPIO);
+#endif
 
 #ifdef CONFIG_DM_REGULATOR
 	ret = regulators_enable_boot_on(false);
-- 
2.25.1

